### ISSUE TITLE
Inconsistent behaviors when "runtime" value is mismatched with function language 

### ISSUE THREAD

**Issue by mrutkows** [apache/incubator-openwhisk-wskdeploy#581]
```yaml
package:
  name: hello_world_package
  version: 1.0
  license: Apache-2.0
  actions:
    hello_world_runtime:
      function: src/hello.js
      runtime: python
```

```json
 "response": {
        "result": {
            "error": "The action failed to generate or locate a binary. See logs for details."
        },

```

if you select "python@2" it works...

**Comment by houshengbo** [comment]
@mrutkows 
function: src/hello.js defines js and runtime points to python????
They are not consistent with each other.

**Comment by mrutkows** [comment]
@houshengbo yes, i am working on testing variations. Please do not attempt to fix yet.  It is indeed a "mismatch", but trying to see what error messages we emit and when we allow them to pass to OpenWhisk (and do not detect).

**Comment by mrutkows** [comment]
However... this worked:
```yaml
package:
  name: hello_world_package
  version: 1.0
  license: Apache-2.0
  actions:
    hello_world_runtime:
      function: src/hello.py
      runtime: nodejs
```

```json
        "result": {
            "greeting": "Hello, stranger from unknown"
        },
```

**Comment by mrutkows** [comment]
It seems we SHOULD be doing some client-side mismatch detection and issue some WARNING (and perhaps allow user to still deploy)...  However, is there a bug or inconsistent behavior on the actual OpenWhisk platform?

**Comment by mrutkows** [comment]
Now when I try
```
package:
  name: hello_world_package
  version: 1.0
  license: Apache-2.0
  actions:
    hello_world_runtime:
      function: src/hello.py
      runtime: java
```
I get an error from wskdeploy:
```
Error: /Users/Matt/go/src/github.com/apache/incubator-openwhisk-wskdeploy/deployers/servicedeployer.go [623]: OpenWhisk Client Error =====> The request content was malformed:
'main' must be a string defined in 'exec' for 'java' actions (code 31062040) Error code: 144.
```

**Comment by pritidesai** [comment]
This was fixed with PR #669.
